onedal/neighbors/tests/test_knn_classification.py

```diff
@@ -0,0 +1,152 @@
+#===============================================================================
+# Copyright 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#===============================================================================
+
+import pytest
+import numpy as np
+from numpy.testing import assert_array_equal, assert_array_almost_equal
+
+from onedal.neighbors import KNeighborsClassifier
+from onedal.tests.utils._device_selection import (get_queues,
+                                                  pass_if_not_implemented_for_gpu)
+
+from sklearn.utils.estimator_checks import check_estimator
+import sklearn.utils.estimator_checks
+from sklearn import datasets
+from sklearn.metrics.pairwise import rbf_kernel
+from sklearn.datasets import make_blobs
+from sklearn.model_selection import train_test_split
+
+
+N_NEIGBORS = 2
+
+
+def _replace_and_save(md, fns, replacing_fn):
+    saved = dict()
+    for check_f in fns:
+        try:
+            fn = getattr(md, check_f)
+            setattr(md, check_f, replacing_fn)
+            saved[check_f] = fn
+        except RuntimeError:
+            pass
+    return saved
+
+
+def _restore_from_saved(md, saved_dict):
+    for check_f in saved_dict:
+        setattr(md, check_f, saved_dict[check_f])
+
+
+def test_estimator():
+    def dummy(*args, **kwargs):
+        pass
+
+    md = sklearn.utils.estimator_checks
+    saved = _replace_and_save(md, [
+        'check_classifiers_predictions',  # could not convert string to float: 'one'
+        'check_classifier_multioutput',  # y should be a 1d array, got an array of shape (42, 3) instead.
+        'check_estimators_unfitted',  # 'KNeighborsClassifier' object has no attribute 'n_samples_fit_'
+        # 'check_sample_weights_invariance',  # Max absolute difference: 0.0008
+        # 'check_estimators_fit_returns_self',  # ValueError: empty metadata
+        # 'check_classifiers_train',  # assert y_pred.shape == (n_samples,)
+        # 'check_estimators_unfitted',  # Call 'fit' with appropriate arguments
+    ], dummy)
+    check_estimator(KNeighborsClassifier())
+    _restore_from_saved(md, saved)

```

Comment:
What actually does this function test?

Reply:
it tests if our onedal estimator belongs to sklearn definition of "estimator". As far as we excluded inheritance of onedal NeighborsBase class from sklearn NeighborsBase, i suppose this kind of tests are no longer valid.

Don't we need to keep this test working how it was done for SVM?

Reply:
I think such kind of tests should be placed in sklearnex module, not in onedal